library/spdm_requester_lib/libspdm_req_challenge.c

```diff
@@ -112,15 +112,19 @@ return_status try_spdm_challenge(IN void *context, IN uint8_t slot_id,
 
     status = spdm_send_spdm_request(spdm_context, NULL,
                     sizeof(spdm_request), &spdm_request);
-    if (RETURN_ERROR(status)) {
+    if (status == RETURN_TIMEOUT) {
+        return status;
+    } else if (RETURN_ERROR(status)) {
         return RETURN_DEVICE_ERROR;
     }
 
     spdm_response_size = sizeof(spdm_response);
     zero_mem(&spdm_response, sizeof(spdm_response));
     status = spdm_receive_spdm_response(
         spdm_context, NULL, &spdm_response_size, &spdm_response);
-    if (RETURN_ERROR(status)) {
+    if (status == RETURN_TIMEOUT) {
+        return status;
+    } else if (RETURN_ERROR(status)) {
```

Comment:
Do we really want to use this pattern?
Can we just make it simple:
```
if (RETURN_ERROR(status)) {
        return status;
}
```

Or, if we really want to do some special handling, why not do it inside of spdm_receive_spdm_response() ?